packages/authenticator/amplify_authenticator/lib/src/widgets/totp/totp_setup_fields.dart

```diff
@@ -0,0 +1,207 @@
+      'Expected TOTP setup uri in state for current screen, instead got $totpUri',
+    );
+    return totpUri!;
+  }
+
+  String getButtonText(ButtonResolverKey key) {
+    return stringResolver.buttons.resolve(
+      context,
+      key,
+    );
+  }
+
+  Future<void> _copyKey() async {
+    setState(() {
+      showKey = true;
+    });
+    try {
+      await Clipboard.setData(
+        ClipboardData(text: state.totpSetupDetails!.sharedSecret),
+      );
+      // There is a bug in the analysis that causes this line to fail linting
+      // This check resolves lint error in beta, thus the lint error can be
+      // ignored using use_build_context_synchronously
+      if (!context.mounted) return;
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          backgroundColor: Theme.of(context).colorScheme.primary,
+          content: const Text(
+            'Copied to clipboard!',
+            textAlign: TextAlign.center,
+          ),
+        ),
+      );
+    } on PlatformException {
+      if (!context.mounted) return;
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          backgroundColor: Theme.of(context).colorScheme.onError,
+          content: const Text(
+            'Copy to clipboard failed.',
+            textAlign: TextAlign.center,
+          ),
+        ),
+      );
+    }
+  }
+
+  // Returns a filled button if the conditional is enabled,
+  // otherwise returns outline button
+  Widget iconToggleButton({
+    required bool conditional,
+    required IconData icon,
+    required String label,
+    required void Function() onPressed,
+  }) {
+    return conditional
+        ? FilledButton.tonalIcon(
+            onPressed: onPressed,
+            icon: Icon(icon),
+            label: Center(
+              child: Text(
+                label,
+              ),
+            ),
+          )
+        : OutlinedButton.icon(
+            onPressed: onPressed,
+            icon: Icon(icon),
+            label: Center(
+              child: Text(
+                label,
+              ),
+            ),
+          );
+  }
+
+  Widget optionDivider() {
+    return const Row(
+      children: [
+        Expanded(
+          child: Divider(height: 0),
+        ),
+        Padding(
+          padding: EdgeInsets.symmetric(horizontal: 10),
+          child: Text(
+            'or',
+            textAlign: TextAlign.center,
+          ),
+        ),
+        Expanded(
+          child: Divider(height: 0),
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget buildFormField(BuildContext context) {
+    final brightness = MediaQuery.of(context).platformBrightness;
+    final isDarkMode = brightness == Brightness.dark;
+    final isMobile = isMobileOrWebMobile(context);
+    const spacingBox = SizedBox(height: 20);
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        /// QR Code Toggle Button
+        if (isMobile) ...[
```

Comment:
This won't show on tablets either. Is that what we want?

Reply:
Yes, we discussed having tablet mirror mobile behavior.